File: src/main/java/ai/ancf/lmos/operator/reconciler/ChannelRolloutReconciler.java
/*
 * SPDX-FileCopyrightText: 2024 Deutsche Telekom AG
 *
 * SPDX-License-Identifier: Apache-2.0
 */

package ai.ancf.lmos.operator.reconciler;

import io.javaoperatorsdk.operator.api.reconciler.Context;
import io.javaoperatorsdk.operator.api.reconciler.Reconciler;
import io.javaoperatorsdk.operator.api.reconciler.UpdateControl;
import ai.ancf.lmos.operator.resources.rollout.ChannelRolloutResource;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;



public class ChannelRolloutReconciler implements Reconciler<ChannelRolloutResource> {

    private static final Logger LOG = LoggerFactory.getLogger(ChannelRolloutReconciler.class);


    @Override
    public UpdateControl<ChannelRolloutResource> reconcile(ChannelRolloutResource channelRolloutResource, Context context) {
        // TODO: fill in logic

        LOG.debug("ChannelRollout reconcile");

        return UpdateControl.noUpdate();
    }
}

File: src/main/kotlin/ai/ancf/lmos/operator/reconciler/ChannelRolloutReconciler.kt
/*
 * SPDX-FileCopyrightText: 2024 Deutsche Telekom AG
 *
 * SPDX-License-Identifier: Apache-2.0
 */

package ai.ancf.lmos.operator.reconciler

import ai.ancf.lmos.operator.resources.ChannelRolloutResource
import io.javaoperatorsdk.operator.api.reconciler.Context
import io.javaoperatorsdk.operator.api.reconciler.Reconciler
import io.javaoperatorsdk.operator.api.reconciler.UpdateControl
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class ChannelRolloutReconciler : Reconciler<ChannelRolloutResource> {
    private val log: Logger = LoggerFactory.getLogger(javaClass)

    override fun reconcile(
        channelRolloutResource: ChannelRolloutResource,
        context: Context<ChannelRolloutResource>,
    ): UpdateControl<ChannelRolloutResource> {
        log.debug("ChannelRollout reconcile")
        return UpdateControl.noUpdate()
    }
}
